app/src/main/java/org/simple/clinic/sync/indicator/SyncIndicatorViewController.kt

```diff
@@ -40,18 +41,39 @@ class SyncIndicatorViewController @Inject constructor(
 
   private fun updateIndicatorView(events: Observable<UiEvent>): Observable<UiChange> {
     val screenCreated = events.ofType<SyncIndicatorViewCreated>()
-    val syncResultStream = lastSyncGroupResult.asObservable()
+    val syncStateStream = lastSyncState
+        .asObservable()
         .distinctUntilChanged()
-        .filterAndUnwrapJust()
-
-    return Observables.combineLatest(screenCreated, syncResultStream) { _, result ->
-      val state = when (result) {
-        SUCCESS -> Synced
-        FAILURE -> SyncPending
-        SYNCING -> Syncing
-      }
-      { ui: Ui -> ui.updateState(state) }
+        .filter { it.lastSyncProgress != null }
+
+    return Observables
+        .combineLatest(screenCreated, syncStateStream)
+        { _, stateStream ->
+          val indicatorState = when (stateStream.lastSyncProgress!!) {
+            SUCCESS -> syncedSuccessState(stateStream.lastSyncSuccessTimestamp)
+            FAILURE -> SyncPending
+            SYNCING -> Syncing
+          }
+          { ui: Ui -> ui.updateState(indicatorState) }
+        }
+  }
+
+  private fun syncedSuccessState(timestamp: Instant?): SyncIndicatorState {
+    if (timestamp == null) {
+      return Synced
+    }
+
+    val now = Instant.now(utcClock)
+    val difference = Math.max(0, now.toEpochMilli() - timestamp.toEpochMilli())
+    val timeSinceLastSync = Duration.ofMillis(difference)
+    val mostFrequentSyncInterval = enumValues<SyncInterval>().map { it.frequency }.min()!!
+
+    val maxDuration = Duration.ofHours(12)
```

Comment:
Magic values like this should be injected via a `Config` object. Can you move this to a config object?

Reply:
Good idea!

Reply:
Do you mind if I do this in the next PR?
It will involve adding a new `SyncIndicatorConfig`.